drivers/pds/dataservice/dataservices.go

```diff
@@ -0,0 +1,332 @@
+}
+
+func (d *DataserviceType) DeployDataservicesAndCreateContext() ([]*scheduler.Context, error) {
+	log.InfoD("Deployment of pds apps called from schedule applications")
+	var deployments = make(map[PDSDataService]*pds.ModelsDeployment)
+	var pdsApps []*pds.ModelsDeployment
+	contexts := make([]*scheduler.Context, 0)
+	var testparams TestParams
+
+	pdsParams := pdslib.GetAndExpectStringEnvVar("PDS_PARAM_CM")
+	params, err := customparams.ReadParams(pdsParams)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read pds params %v", err)
+	}
+	infraParams := params.InfraToTest
+	namespace := params.InfraToTest.Namespace
+
+	_, isAvailable, err := pdslib.CreatePDSNamespace(namespace)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create pds namespace %v", err)
+	}
+	if !isAvailable {
+		return nil, fmt.Errorf("pdsnamespace %v is not available to deploy apps", namespace)
+	}
+	_, tenantID, dnsZone, projectID, serviceType, clusterID, err := pdslib.SetupPDSTest(infraParams.ControlPlaneURL, infraParams.ClusterType,
+		infraParams.AccountName, infraParams.TenantName, infraParams.ProjectName)
+	testparams.ServiceType = serviceType
+	if err != nil {
+		return nil, fmt.Errorf("Failed on SetupPDSTest method %v", err)
+	}
+	testparams.DnsZone = dnsZone
+
+	deploymentTargetID, err = pdslib.GetDeploymentTargetID(clusterID, tenantID)
+	log.FailOnError(err, "Failed to get the deployment TargetID")
+	log.InfoD("DeploymentTargetID %s ", deploymentTargetID)
+	testparams.DeploymentTargetId = deploymentTargetID
+
+	namespaceId, err := pdslib.GetnameSpaceID(namespace, deploymentTargetID)
+	log.FailOnError(err, "Failed to get the namespace Id")
+	log.InfoD("NamespaceId %s ", namespaceId)
+	testparams.NamespaceId = namespaceId
+
+	storageTemplateID, err = pdslib.GetStorageTemplate(tenantID)
+	log.FailOnError(err, "Failed while getting storage template ID")
+	log.InfoD("storageTemplateID %v", storageTemplateID)
+	testparams.StorageTemplateId = storageTemplateID
+
+	for _, ds := range params.DataServiceToTest {
+		deployment, _, _, err := d.TriggerDeployDataService(ds, namespace, tenantID, projectID, false, testparams)
+		if err != nil {
+			return nil, fmt.Errorf("failed to deploy pds apps %v", err)
+		}
+		deployments[ds] = deployment
+		pdsApps = append(pdsApps, deployment)
+	}
+
+	log.InfoD("Creating Context for PDS Apps")
+	contexts = d.CreateAppContext(pdsApps)
+	return contexts, nil
+}
+
+func (d *DataserviceType) CreateAppContext(pdsApps []*pds.ModelsDeployment) []*scheduler.Context {
+	var specObjects []interface{}
+	var Contexts []*scheduler.Context
+	var ctx *scheduler.Context
+
+	for _, dep := range pdsApps {
+		specObjects = append(specObjects, dep)
+		ctx = &scheduler.Context{
+			UID: dep.GetId(),
+			App: &spec.AppSpec{
+				Key:      *dep.ClusterResourceName,
+				SpecList: specObjects,
+			},
+		}
+		Contexts = append(Contexts, ctx)
+	}
+	return Contexts
+}
+
+func DataserviceInit(ControlPlaneURL string) (*DataserviceType, error) {
```

Comment:
please use InitializeApiComponents to avoid duplication.

Reply:
done